worlds/lingo/static_logic.py

```diff
@@ -0,0 +1,512 @@
+                    self.DOOR_LOCATION_IDS[room_name] = {}
+                    self.DOOR_ITEM_IDS[room_name] = {}
+
+                    for door_name, door_data in config["doors"][room_name].items():
+                        if "location" in door_data:
+                            self.DOOR_LOCATION_IDS[room_name][door_name] = door_data["location"]
+
+                        if "item" in door_data:
+                            self.DOOR_ITEM_IDS[room_name][door_name] = door_data["item"]
+
+            if "door_groups" in config:
+                for item_name, item_id in config["door_groups"].items():
+                    self.DOOR_GROUP_ITEM_IDS[item_name] = item_id
+
+            if "progression" in config:
+                for item_name, item_id in config["progression"].items():
+                    self.PROGRESSIVE_ITEM_IDS[item_name] = item_id
+
+        with files("worlds.lingo").joinpath("LL1.yaml").open() as file:
+            config = yaml.load(file, Loader=yaml.Loader)
+
+            for room_name, room_data in config.items():
+                self.process_room(room_name, room_data)
+
+            self.PAINTING_EXITS = len(self.PAINTING_EXIT_ROOMS)
+
+    def get_special_item_id(self, name: str):
+        if name not in self.SPECIAL_ITEM_IDS:
+            raise Exception(f'Item ID for special item {name} not found in ids.yaml.')
+
+        return self.SPECIAL_ITEM_IDS[name]
+
+    def get_panel_location_id(self, room: str, name: str):
+        if room not in self.PANEL_LOCATION_IDS or name not in self.PANEL_LOCATION_IDS[room]:
+            raise Exception(f'Location ID for panel {room} - {name} not found in ids.yaml.')
+
+        return self.PANEL_LOCATION_IDS[room][name]
+
+    def get_door_location_id(self, room: str, name: str):
+        if room not in self.DOOR_LOCATION_IDS or name not in self.DOOR_LOCATION_IDS[room]:
+            raise Exception(f'Location ID for door {room} - {name} not found in ids.yaml.')
+
+        return self.DOOR_LOCATION_IDS[room][name]
+
+    def get_door_item_id(self, room: str, name: str):
+        if room not in self.DOOR_ITEM_IDS or name not in self.DOOR_ITEM_IDS[room]:
+            raise Exception(f'Item ID for door {room} - {name} not found in ids.yaml.')
+
+        return self.DOOR_ITEM_IDS[room][name]
+
+    def get_door_group_item_id(self, name: str):
+        if name not in self.DOOR_GROUP_ITEM_IDS:
+            raise Exception(f'Item ID for door group {name} not found in ids.yaml.')
+
+        return self.DOOR_GROUP_ITEM_IDS[name]
+
+    def get_progressive_item_id(self, name: str):
+        if name not in self.PROGRESSIVE_ITEM_IDS:
+            raise Exception(f'Item ID for progressive item {name} not found in ids.yaml.')
+
+        return self.PROGRESSIVE_ITEM_IDS[name]
+
+    def process_entrance(self, source_room, doors, room_obj):
+        if doors is True:
+            room_obj.entrances.append(RoomEntrance(source_room, None, False))
+        elif isinstance(doors, Dict):
```

Comment:
It looks like this technically works, but I would guess that `isinstance(doors, dict)` is probably more correct and safer.